### PR TITLE
Add bazam and biobambam to stream realignment CRAM to BAM

### PR DESCRIPTION
**Pull request by vladsavelyev**
Update the `WholeGenomeGermlineSingleSample` workflow to use [Bazam](https://github.com/ssadedin/bazam) and [biobambam](https://github.com/gt1/biobambam2) to process the input BAM/CRAM into a realigned and sorted/dedupped CRAM just in one step.

Removed the input BAM sharding feature, so the alignment is only parallelised with threads on a single machine. It allows to avoid a lot of overhead copying alignment around and merging, while keeping the run time within 8 hours.

Also added a feature to run a test on one chromosome.

Will put updated input templates into cromwell-config.

### CODE REVIEW

File: tasks/broad/Alignment.wdl [hunk 1]
```diff
@@ -17,6 +17,76 @@ version 1.0
 
 import "../../structs/dna_seq/DNASeqStructs.wdl"
 
+task Bazam {
+  input {
+    File input_bam
+    String bwa_commandline
+    String output_bam_basename
```
Comment (lgruen) [inline tasks/broad/Alignment.wdl]: What would you think of reading and writing CRAMs directly?

**Review by lgruen — CHANGES_REQUESTED** [review]
Nice!!

File: tasks/broad/Alignment.wdl (hunk 1 above)
Comment (vladsavelyev) [reply tasks/broad/Alignment.wdl]: Definitely can do that, yes!

File: tasks/broad/Alignment.wdl [hunk 2]
```diff
@@ -17,6 +17,86 @@ version 1.0
...
+  
+  runtime {
+    # docker: "australia-southeast1-docker.pkg.dev/fewgenomes/images/bazam:v2"
+    # cromwell doesn't work with artifact registry:
+    # java.lang.Exception: Registry australia-southeast1-docker.pkg.dev is not supported
+    docker: "gcr.io/fewgenomes/bazam:v2"
+    preemptible: preemptible_tries
+    memory: "64 GiB"
+    cpu: total_cpu
+    disks: "local-disk " + 500 + " HDD"
```
Comment (lgruen) [inline tasks/broad/Alignment.wdl]: Nice -- this seems more robust than a multiplier based on BAM vs CRAM and is probably only slightly more expensive.

File: tasks/broad/GermlineVariantDiscovery.wdl [hunk 3]
```diff
@@ -92,7 +92,7 @@ task HaplotypeCaller_GATK4_VCF {
     Boolean make_bamout
     Int preemptible_tries
     Int hc_scatter
-    String gatk_docker = "us.gcr.io/broad-gatk/gatk:4.1.4.0"
+    String gatk_docker = "us.gcr.io/broad-gatk/gatk:4.2.0.0"
```
Comment (lgruen) [inline tasks/broad/GermlineVariantDiscovery.wdl:95]: Was this change intentional? Isn't KCCG still using an earlier version?

**Review by lgruen — APPROVED** [review]
How difficult would it be to also support FASTQ inputs?

File: tasks/broad/GermlineVariantDiscovery.wdl (hunk 3 above)
Comment (vladsavelyev) [reply tasks/broad/GermlineVariantDiscovery.wdl:95]: Yeah, intentional. The thing is that sharding the variant calling from CRAMs is not working with 4.1.4.0, when an interval spans 2 chromosomes.

Example:

```
gatk --java-options "-Xms4000m -XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10"   HaplotypeCaller   -R /hg38/Homo_sapiens_assembly38.fasta  -I gs://playground-au/cromwell/NA12878-full.cram  -L 40scattered-chr15-chr16.interval_list   -O test.g.vcf.gz   -contamination 0   -G StandardAnnotation -G StandardHCAnnotation -G AS_StandardAnnotation   -GQB 10 -GQB 20 -GQB 30 -GQB 40 -GQB 50 -GQB 60 -GQB 70 -GQB 80 -GQB 90   -ERC GVCF
```

4.1.4.0 here fails `when 40scattered-chr15-chr16.interval_list` spans chr15 and chr16:

```
...
chr15   101900000       101981189       +       . intersection ACGTmer.177
chr16   10001   99999   +       . intersection ACGTmer.1
```

4.1.4.0 works however if the interval list only contains intervals from a single chromosome.

4.2.0.0 works nicely either way.

File: tasks/broad/Alignment.wdl (hunk 2 above)
Comment (vladsavelyev) [reply tasks/broad/Alignment.wdl]: Actually reduced to 300G, seems to be enough.

**Comment by vladsavelyev** [comment]
Supporting fastq inputs now.
